src/Microsoft.IdentityModel.Tokens/Base64UrlEncoder.cs

```diff
@@ -73,10 +84,84 @@ public static string Encode(string arg)
         /// <exception cref="ArgumentOutOfRangeException">offset or length is negative OR offset plus length is greater than the length of inArray.</exception>
         public static string Encode(byte[] inArray, int offset, int length)
         {
-            if (inArray == null)
-                throw LogHelper.LogArgumentNullException("inArray");
+            _ = inArray ?? throw LogHelper.LogArgumentNullException("inArray");
+
+            if (length < 0)
+            {
+                throw LogHelper.LogArgumentException<ArgumentException>(nameof(length), $"{nameof(length)} must be greater or equal than zero");
```

Comment:
why does a method named 'LogArgumentException' need to be generic?
why the <ArgumentException> ?